Week3/homework/step2-6.js

```diff
@@ -4,15 +4,37 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArr = [];
+  arr.forEach(element => {
+    element.forEach(item => {
+      newArr.push(item);
+    });
+  });
+  return newArr;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArr = [];
+  arr.forEach(element => {
+    element.forEach(item => {
+      item.forEach(unit => {
+        newArr.push(unit);
+      });
+    });
+  });
+  return newArr;
 }
 
+/* this can be achieved also by using flat() method.
+  function flattenArray2d(){
+  return arr2d.flat(); // flattens down 1 dimension
+  }
+
+  function flattenArray3d(){
+  return arr3d.flat(2); // flattens down 2 dimension
+  }
+*/
+
```

Comment:
Quite nice. I wasn't aware of `flat` method before. A good idea to share this with your fellow students and mentors like me who weren't aware of it in the Slack channel. 😜